### PR TITLE
update constructor lesson to include visualization, add bullet list o…

### PR DESCRIPTION
**Pull request by linda-seiter**
…f rules, signature images

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,31 +2,365 @@
...
+- If you *do not* define a constructor, Java implicitly defines a default no-args constructor.
+- If you *do* define a constructor with or without parameters, Java *does not* generate a default constructor.
+
+The default constructor for `Dog` initializes instance variables to default values
+based on their data type:
+
+```java
+
+// Constructor
+public Dog() {
```
Comment (kcleland0818) [inline README.md]: Suggestion: Show the private instance variables above too or show the `Dog` class again in the introduction

**Review by kcleland0818 — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -2,31 +2,365 @@
...
+// Constructor #1
+public Dog(String name, int age) {
+    this.name = name;
+    this.age = age;
+    this.breed = "unknown";
+}
+
+// Constructor #2
+public Dog(String name, String breed, int age, boolean waggingTail) {
+    this(name, age);  // calls Constructor #1
```
Comment (developerjaym) [inline README.md]: Consider having constructor 1 call constructor 2
```
public Dog(String name, int age) {
 this(name, "Unknown", age, false);
}
```

**Review by developerjaym — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (linda-seiter) [reply README.md]: ok

File: README.md (hunk 2 above)
Comment (linda-seiter) [reply README.md]: yes, that is much better!

File: README.md [hunk 3]
```diff
@@ -2,31 +2,428 @@
...
+public class Dog {
+
+    private String name;
+    private String breed;
+    private int age;
+    private boolean waggingTail;
+
+    // Constructor #1
+    public Dog(String name, String breed, int age, boolean waggingTail) {
+        this(name, age); //calls Constructor #2
```
Comment (developerjaym) [inline README.md]: Should the final version of the `Dog` class have constructor 2 call constructor 1?

Comment (linda-seiter) [reply README.md]: Yes, thanks!

**Review by kcleland0818 — APPROVED** [review]

**Review by developerjaym — APPROVED** [review]

**Review by AlveeM — APPROVED** [review]

**Pull request merged**